src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/FileLoggerProvider.cs

```diff
@@ -4,69 +4,78 @@
+        {
+            if (outputTemplate == null)
             {
-                if (string.IsNullOrEmpty(webAppConfiguration.FileLoggingFolder))
-                {
-                    throw new ArgumentNullException(nameof(webAppConfiguration.FileLoggingFolder), "The file logger path cannot be null or empty.");
-                }
-
-                var logsFolder = webAppConfiguration.FileLoggingFolder;
-                if (!Directory.Exists(logsFolder))
-                {
-                    Directory.CreateDirectory(logsFolder);
-                }
-                var logsFilePattern = Path.Combine(logsFolder, FileNamePattern);
-
-                var fileSizeLimitBytes = fileSizeLimit * 1024 * 1024;
-
-                var messageFormatter = new MessageTemplateTextFormatter(OutputTemplate, null);
-                var rollingFileSink = new RollingFileSink(logsFilePattern, messageFormatter, fileSizeLimitBytes, RetainedFileCountLimit);
-                var backgroundSink = new BackgroundSink(rollingFileSink, BackgroundSink.DefaultLogMessagesQueueSize);
+                throw new ArgumentNullException(nameof(outputTemplate));
+            }
+            if (fileSizeLimit <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(fileSizeLimit), $"{nameof(fileSizeLimit)} should be positive.");
+            }
+            if (retainedFileCountLimit <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(retainedFileCountLimit), $"{nameof(retainedFileCountLimit)} should be positive.");
+            }
 
-                loggerConfiguration.WriteTo.Sink(backgroundSink);
-            })
-        {
+            _fileSizeLimit = fileSizeLimit;
+            _retainedFileCountLimit = retainedFileCountLimit;
+            _outputTemplate = outputTemplate;
         }
 
         /// <inheritdoc />
-        protected override void OnConfigurationChanged(WebAppLogConfiguration newConfiguration)
+        public Logger ConfigureLogger(IWebAppLogConfigurationReader reader)
         {
-            if (!newConfiguration.FileLoggingEnabled)
+            var webAppConfiguration = reader.Current;
+            if (string.IsNullOrEmpty(webAppConfiguration.FileLoggingFolder))
             {
-                LevelSwitcher.MinimumLevel = LogLevelDisabled;
+                throw new ArgumentNullException(nameof(webAppConfiguration.FileLoggingFolder),
+                    "The file logger path cannot be null or empty.");
             }
-            else
+
+            var logsFolder = webAppConfiguration.FileLoggingFolder;
+            if (!Directory.Exists(logsFolder))
             {
-                LevelSwitcher.MinimumLevel = LogLevelToLogEventLevel(newConfiguration.FileLoggingLevel);
+                Directory.CreateDirectory(logsFolder);
             }
+            var logsFilePattern = Path.Combine(logsFolder, FileNamePattern);
```

Comment:
Probably this should include the instance id too because the network share is shared between instances